### PR TITLE
Ruby implementation of xcproj functionality

### PR DESCRIPTION
**Pull request by neonichu**
see #199

### CODE REVIEW

**Comment by segiddins** [comment]
@neonichu ping

**Comment by neonichu** [comment]
@segiddins IDK, seems to me that there is little interest in merging this, so probably should just be closed.

**Comment by kylef** [comment]
@neonichu There is certainly interest in getting this into master. We've been waiting for you to rebase this branch before it's even mergable.

screen shot 2015-01-12 at 16 39 14

Along with addressing the outstanding comments.

**Comment by segiddins** [comment]
Well done @neonichu! Let's review this one more time, and then get this into the final 0.36 release!

**Comment by neonichu** [comment]
@kylef sorry, I thought I did that already, but turns out I merged master into this instead of doing a rebase.  ¯\_(ツ)_/¯

File: lib/xcodeproj/plist_helper.rb [hunk 1]
```diff
@@ -70,6 +76,24 @@ def file_in_conflict?(path)
       ensure
         file.close
       end
+
+      def ruby_hash_write_xcode(hash, path)
```
Comment (kylef) [inline lib/xcodeproj/plist_helper.rb]: @neonichu Could you add some documentation to this stating which type this method expects. I.e, `path` being a pathname or string etc.

File: lib/xcodeproj/plist_helper.rb [hunk 2]
```diff
@@ -450,3 +478,166 @@ def self.RubyBooleanToCFBoolean(value)
...
+    ID = CoreFoundation::VoidPointer
+    SEL = CoreFoundation::VoidPointer
+
+    extern :NSSelectorFromString, [CoreFoundation::CFTypeRef], SEL
+
+    extern :objc_getClass, [CoreFoundation::CharPointer], Class
+    extern :class_getName, [Class], CoreFoundation::CharPointer
+  end
+
+  XCODE_PATH = Pathname.new(`xcrun xcode-select -p`.strip).dirname
```
Comment (kylef) [inline lib/xcodeproj/plist_helper.rb]: `xcrun xcode-select -p` won't return the path in some cases, for example:

- Xcode not being installed
- User not yet agreed to license
    ```
    $ xcrun xcode-select -p


    Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.
    ```

I'm wondering if we need to change/account for this in anyway.

Comment (neonichu) [inline lib/xcodeproj/plist_helper.rb]: I'll add a spec for this. IMHO, we should silently fallback to XML writing, just as in other cases of not being able to use the private Xcode API.

Comment (neonichu) [inline lib/xcodeproj/plist_helper.rb]: Added in 96699428955b0550f9808e7ed97c71939f53d2ff

**Comment by orta** [comment]
awesome

**Pull request merged**